components/menuData.ts

```diff
@@ -1547,7 +1555,76 @@ menuDataRouter.post(
                     HasPrevious: false,
                     HasMore: false,
                 },
-            },
+            }
+        } else {
+            // No plugins handle this. Getting search results from official
+            searchResult = await officialSearchContract(
+                req.jwt.unique_name,
+                req.gameVersion,
+                req.body,
+                0,
+            )
+        }
+
+        res.json({
+            template: getVersionedConfig(
+                "ContractSearchResponseTemplate",
+                req.gameVersion,
+                false,
+            ),
+            data: searchResult,
+        })
+    },
+)
+
+async function officialSearchContract(
```

Comment:
This file is too big. We should move this elsewhere.

Reply:
To where? This function is only called by two other functions in this file...

Reply:
Somewhere else, I'm just tired of seeing this file get longer and longer.

Reply:
Moved to `contractsModeRouting.ts`.